src/prefect/infrastructure/kubernetes.py

```diff
@@ -625,24 +641,23 @@ def _watch_job(self, job_name: str) -> int:
                         exc_info=True,
                     )
 
-        self.logger.debug(f"Job {job_name!r}: Starting watch for job completion")
-        deadline = (
-            (time.time() + self.job_watch_timeout_seconds)
-            if self.job_watch_timeout_seconds is not None
-            else None
-        )
-        completed = False
-        while not completed:
-            remaining_time = deadline - time.time() if deadline else None
-            if deadline and remaining_time <= 0:
-                self.logger.error(
-                    f"Job {job_name!r}: Job did not complete within "
-                    f"timeout of {self.job_watch_timeout_seconds}s."
-                )
-                return -1
+        with self.get_batch_client() as batch_client:
+            # Check if the job is completed before beginning a watch
+            job = batch_client.read_namespaced_job(
+                name=job_name, namespace=self.namespace
+            )
+            completed = job.status.completion_time is not None
```

Comment:
Here we check the status _before_ starting the watch so that if a deadline is exceeded during log streaming but the job completes then we do not fail the job as timed out.